portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -42,37 +45,64 @@ public DataServlet() {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {  
     // Get comments from Datastore and show on the page.
-    ArrayList<String> messages = new ArrayList<String>();
-    Query query = new Query("Comment").addSort("text", SortDirection.DESCENDING);
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    Query query = new Query("Comment").addSort("sentiment", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
-        String comment = (String) entity.getProperty("text");
-        messages.add(comment);
+        String comment_text = (String) entity.getProperty("text");
+        double sentiment_score = (double) entity.getProperty("sentiment");
+        Comment comment = new Comment(comment_text, sentiment_score);
+        comments.add(comment);
     }
 
-    String json = convertToJsonUsingGson(messages);
+    String json = convertToJsonUsingGson(comments);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
 
   /**
-   * Converts a Java ArrayList<String> into a JSON string using the Gson library.
+   * Converts a Java ArrayList<Comment> into a JSON string using the Gson library.
    */
-  private String convertToJsonUsingGson(ArrayList<String> listOfStrings) {
+  private String convertToJsonUsingGson(ArrayList<Comment> listOfStrings) {
     String json = gson.toJson(listOfStrings);
     return json;
   }
 
+  /**
+   * Class for comments.
+   */
+  private static class Comment {
+      String text;
+      double sentiment_score;
+
+      public Comment(String text, double sentiment_score) {
+        this.text = text;
+        // truncate the double
+        int temp = (int)(sentiment_score*100);
+        double truncated = temp/100d;
+        this.sentiment_score = truncated;
+      }
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the comment input from the form.
     String comment = getParameter(request, "text-input", "");
 
-    // Add the comment to Datastore.
+    // Calculate sentiment score.
+    Document doc =
+        Document.newBuilder().setContent(comment).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
```

Comment:
Consider initializing this as member variable of the class, instead of recreating it every time the method is called. This could make it faster, but you wouldn't be able to call `languageService.close()`.